packages/dashboard/src/customization/widgets/barChart/component.tsx

```diff
@@ -33,6 +34,27 @@ const BarChartWidgetComponent: React.FC<BarChartWidget> = (widget) => {
   const aggregation = getAggregation(queryConfig);
 
   const significantDigits = widgetSignificantDigits ?? dashboardSignificantDigits;
+  const grid = useSelector((state: DashboardState) => state.grid);
+  const size = { width: grid.cellSize * widget.width, height: grid.cellSize * widget.height };
+
+  if (showEcharts) {
+    return (
+      <Chart
```

Comment:
we should probably have a generic widget component and this component might not be even required...

Reply:
we could split each out into a separate file in the utils folder, or move the TC/interpolation helpers closer to where they're used